src/cairoUtilFuncGen/storage/dynArrayPop.ts

```diff
@@ -88,31 +89,27 @@ export class DynArrayPopGen extends StringIndexedFuncGen {
 
     const getElemLoc =
       isDynamicArray(elementType) || isMapping(elementType)
-        ? [
-            `let (elem_loc) = ${arrayName}.read(loc, newLen);`,
-            `let (elem_loc) = readId(elem_loc);`,
-          ].join('\n')
-        : `let (elem_loc) = ${arrayName}.read(loc, newLen);`;
+        ? endent`
+            let elem_loc = ${arrayName}::read((loc, newLen));
+            let elem_loc = readId(elem_loc);
+            `
```

Comment:
Does double definition of the same variable work in Cairo 1?

Reply:
yes